test/core/crawlers/file_download.test.ts

```diff
@@ -0,0 +1,167 @@
+import { pipeline } from 'node:stream/promises';
+import { ReadableStream } from 'node:stream/web';
+import { setTimeout } from 'node:timers/promises';
+
+import { Configuration, FileDownload } from '@crawlee/http';
+
+class ReadableStreamGenerator {
+    private static async generateRandomData(size: number, seed: number) {
+        const chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+        const buffer = Buffer.alloc(size);
+        for (let i = 0; i < size; i++) {
+            // eslint-disable-next-line no-bitwise
+            seed = Math.imul(48271, seed) | 0 % 2147483647;
+            buffer[i] = chars.charCodeAt(seed % chars.length);
+        }
+        return buffer;
+    }
+
+    static getReadableStream(size: number, seed: number, throttle: number = 0): ReadableStream {
+        let bytesRead = 0;
+        const stream = new ReadableStream({
+            start: async (controller) => {
+                while (bytesRead < size) {
+                    const chunkSize = Math.min(size - bytesRead, 1024);
+                    const chunk = await this.generateRandomData(chunkSize, seed);
+                    bytesRead += chunk.length;
+                    controller.enqueue(chunk);
+
+                    if (throttle > 0) {
+                        await setTimeout(throttle);
+                    }
+                }
+                controller.close();
+            },
+        });
+
+        return stream;
+    }
+
+    static async getBuffer(size: number, seed: number) {
+        const stream = this.getReadableStream(size, seed);
+        const chunks: string[] = [];
+        for await (const chunk of stream) {
+            chunks.push(chunk);
+        }
+        return Buffer.from(chunks.join(''));
+    }
+}
+
+const router = new Map<string, http.RequestListener>();
+router.set('/file', async (req, res) => {
+    const url = new URL(req.url, 'http://localhost');
+
+    const size = Number(url.searchParams.get('size') ?? 1024);
+    const seed = Number(url.searchParams.get('seed') ?? 123);
+    const throttle = Number(url.searchParams.get('throttle') ?? 0);
+
+    const stream = ReadableStreamGenerator.getReadableStream(size, seed, throttle);
+
+    res.setHeader('content-type', 'application/octet-stream');
+    await pipeline(stream, res);
+
+    res.end();
+});
+
+let server: http.Server;
+let url: string;
+
+beforeAll(async () => {
+    server = http.createServer((request, response) => {
+        try {
+            const requestUrl = new URL(request.url, 'http://localhost');
+            router.get(requestUrl.pathname)(request, response);
+        } catch (error) {
+            response.destroy();
+        }
+    });
+
+    await new Promise<void>((resolve) => server.listen(() => {
+        url = `http://127.0.0.1:${(server.address() as AddressInfo).port}`;
+        resolve();
+    }));
+});
+
+afterAll(async () => {
+    await new Promise((resolve) => server.close(resolve));
+});
```

Comment:
I mean, this seems OK at a glance, but couldn't you do a similar thing with `nock` or `express`? Both are already being used in tests, and this seems like one more way to do the same thing.